quic/s2n-quic-transport/src/recovery/manager.rs

```diff
@@ -571,6 +577,14 @@ impl<Config: endpoint::Config> Manager<Config> {
                     random_generator,
                     timestamp,
                 );
+                if slow_start && !path.congestion_controller.is_slow_start() {
+                    let path_id = acked_packet_info.path_id;
+                    publisher.on_slow_start_exited(event::builder::SlowStartExited {
+                        path: path_event!(path, path_id),
+                        cause: SlowStartExitCause::Other,
+                        congestion_window,
+                    });
```

Comment:
instead of `Other` can we attach a more meaningful reason. Actually cant tell why the exit happens here? Is it threshold related or loss detected via missing range?

Reply:
Currently the only way slow start would be exited here is if Hybrid Slow Start determined the RTT times were increasing beyond a threshold (or if this is the second time we're in Slow Start and the original slow start threshold had been exceeded). With other congestion controllers there might be other reasons, so thats why I just grouped it under "Other". I didn't feel it made sense to pass the event publisher through to each congestion controller implementation, so I think this is the best we can do here for now.

Reply:
One thing I could do that might help is have Cubic transition out of SlowStart in the `on_rtt_update` method, and then check in the recovery manager if Slow start was exited while updating the RTT. Then I could add a `SlowStartExitCause::Rtt`, to make it more explicit

Reply:
Tried that out in https://github.com/aws/s2n-quic/pull/1373/commits/230bfd6427e256c502679aed92202e955be25c97. I think I like it better since if you don't care about the `SlowStartExited` events that occur the 2nd time on a connection, you can just filter out SlowStartExitCause::Other

Reply:
Once we add `Other` we can never get rid of it. If we know that this case wont get called and is covered by other cases then maybe we leave it out all together.

I dont think BBR has a concept of slow start per say, but other CC might. I wonder if some of these events are Cubic specific. Comments might be necessary for the specific event case.

Reply:
`Other` will be emitted in one scenario currently:
- The connection experiences "persistent congestion", which drops the congestion window to the minimum, below the previously determined slow start threshold, and sets the state to `SlowStart`
- The connection then grows the congestion window, and the previously determined slow start threshold is exceeded, causing SlowStart to exit

BBR has a state similar to slow start that is just called "Startup". It exits based on bandwidth plateauing or excessive loss, though both of these are only checked in `on_ack`. The recovery manager wouldn't be able to tell which one caused the slow start exit, so as it is this would fall under `Other`

I think `Other` should be OK, since the Recovery Manager generally doesn't know the inner workings of the congestion controller. Only in the cases of SlowStart exiting due to RTT, PacketLoss, or ECN signals is it obvious to the Recovery Manager what the cause was. I can add comments to each Cause though explaining it